07-routing-gossip.md

```diff
@@ -285,6 +285,7 @@ The following `address descriptor` types are defined:
          onion service addresses; Encodes:
          `[32:32_byte_ed25519_pubkey] || [2:checksum] || [1:version]`, where
          `checksum = sha3(".onion checksum" | pubkey || version)[:2]`.
+   * `5`: DNS hostname; data = `[byte:len][len*byte:hostname][u16:port]` (length up to 258)
```

Comment:
Limits the length of the hostname to 255 bytes. If this is a purposeful limitation it should be documnted. FWIW I do agree that this should be the limit given that this is a gossip message that is forwarded and stored by every node on the network (even the ones that don't understand the value).

In other places we'd use varint for variable length field prefixes, but in this case I agree with the choice of the single prefix-byte.

Reply:
Yes exactly, Using u16 is too big for DNS hostnames that can never exceed 255 bytes (or even 253).
We could think about adding basetype `u8` but then it would only be used here (for now).
Also we should not send a fixed array of 255bytes as that would be network bloat.

Reply:
For consistency's sake, I would write it like this:

```suggestion
   * `5`: DNS hostname; data = `[1:hostname_len][hostname_len:hostname][2:port]` (length up to 258)
```

But I don't feel strongly about that if you prefer your version.

Reply:
Yep, applied that to have consistency in this section, though throughout the spec the notation `u16` seems more common.